src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -294,14 +294,20 @@ updatePositionMap uri changes = pluginGetFile "updatePositionMap: " uri $ \file
     cacheInfoNoClear file info'
     return $ IdeResultOk ()
   where
-    f (+/-) (J.Range (Position sl _) (Position el _)) txt p@(Position l c)
+    f (+/-) (J.Range (Position sl sc) (Position el ec)) txt p@(Position l c)
       | l < sl = Just p
       | l > el = Just $ Position l' c
+      | l == sl && c <= sc = Just p
+      | l == sl && l == el && c <= nec && newL == 0 = Just $ Position l ec
+      | l == sl && l == el && c > nec && newL == 0 = Just $ Position l (c +/- (nec - sc))
```

Comment:
Can you add a comment on what each case is covering?

Reply:
Added some ASCII diagrams!